tests/tensor/test_math.py

```diff
@@ -2076,6 +2079,86 @@ def is_super_shape(var1, var2):
                             assert is_super_shape(y, g)
 
 
+class TestMatrixVectorOps:
```

Comment:
Tests are still overly-complicated. Here is what I would do. Create one test for the 3 new functions, and just call it once with two matrices, that will always have at least one batch dimension.

Compare with the respective numpy function. Because we test on numpy<2.0 in some jobs of the CI and a `pytest.skip` based on the numpy version.